app/src/main/java/com/nervousfish/nervousfish/modules/pairing/AndroidBluetoothService.java

```diff
@@ -42,20 +43,26 @@
             UUID.fromString("2d7c6682-3b84-4d00-9e61-717bac0b2643");
     // Name for the SDP record when creating server socket
     static final String NAME_SECURE = "BluetoothChatSecure";
+
     private static final Logger LOGGER = LoggerFactory.getLogger("AndroidBluetoothHandler");
     // Binder given to clients
-    private final IBinder mBinder = new LocalBinder();
-    private AndroidBluetoothConnectThread connectThread;
+    @SuppressWarnings("ThisEscapedInObjectConstruction")
+    private final IBinder mBinder = new AndroidBluetoothService.LocalBinder(this);
+    private final Object lock = new Object();
     private BluetoothState state = STATE_NONE;
+    @Nullable
+    private AndroidBluetoothConnectThread connectThread;
```

Comment:
While you're working on this, may I ask why we need to have a separate variable for each Thread kind? Isn't it true that we can have only one of these threads running at any point in time?